src/components/marketplace/order_details.tsx

```diff
@@ -70,18 +90,120 @@ class OrderDetails extends React.Component<Props, State> {
     // TODO: Refactor with hooks (needs react version update)
 
     public state = {
-        orderDetailType: OrderDetailsType.Eth,
-        zeroXPriceInUSD: new BigNumber(0),
+        limitOrder: {
+            orderDetailType: OrderDetailsType.Eth,
+            zeroXFeeInUSD: new BigNumber(0),
+            zeroXFeeInWeth: new BigNumber(0),
+            totalCostInWeth: new BigNumber(0),
+            totalCostInUSD: new BigNumber(0),
+        },
+        marketOrder: {
+            orderDetailType: OrderDetailsType.Eth,
+            zeroXFeeInUSD: new BigNumber(0),
+            zeroXFeeInWeth: new BigNumber(0),
+            totalCostInWeth: new BigNumber(0),
+            totalCostInUSD: new BigNumber(0),
+        },
+    };
+
+    public updateLimitOrderState = async () => {
+        const { tokenAmount, tokenPrice, selectedToken } = this.props;
+        if (selectedToken) {
+            /* Reduces decimals of the token amount */
+            const tokenAmountConverted = tokenAmountInUnitsToBigNumber(tokenAmount, selectedToken.decimals);
+            /* This could be refactored with promise all  */
+            const promisesArray = [getZeroXPriceInWeth(), getZeroXPriceInUSD(), getEthereumPriceInUSD()];
```

Comment:
Q: Am I missing something here or does this mean that we are going to get all the ZRX orders from the relayer every time the user presses a key?

Even if that's not a problem, shouldn't we handle the ZRX/wETH price the same way we handle the ETH/USD price: asking for it in regular intervals and having it as part of the global state?

Reply:
It's now done as a set interval cache, we should add it on the store, maybe on the issue #77 ? @mariano-aguero